### PR TITLE
Proper OpenAPI spec and improved API Explorer

### PR DESCRIPTION
**Pull request by codydbentley**
## Description

- Removed swagger dependencies
- Converted spec to proper yaml spec that is valid
- Refactored ApiExplorer view into bh-shared-ui lib
- Replaced swagger-ui with redoc
- Audited all endpoints for accuracy and updated documentation
- Fixed numerous bugs found in endpoint logic (ex- invalid filters for some entities)
- Refactored AD entity info and AD related entity handlers to reduce code duplication

## Motivation and Context

Our API spec is currently a bit messy and incomplete. We target the OpenAPI Spec (OAS), but our current spec as written is not valid, consisting of what appears to be a frankenstein of Swagger 2.0 and OpenAPI 3.0 specs. We also do not have a proper entry point file (it is currently hard-coded into our swagger server Go file). These problems combined mean that were are putting in the effort to maintain an OAS, but not reaping all the benefits from it, such as code generation.

Fixing our API spec will allow us to gain major benefits of using it beyond just serving the API Explorer. It will also allow our IDEs and editors to give autocomplete suggestions and validation at edit time. We could leverage code generation that could replace large portions of code we currently maintain manually (code that is also incomplete and inconsistent). Users would also be able to use our spec to generate API clients in the language of their choice.

## How Has This Been Tested?

All unit tests and integration tests are passing.
ReDoc is hooked up and new API Explorer works great.

## Types of changes

-   [x] Chore (a change that does not modify the application functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: packages/go/spec/docs.go
Comment (juggernot325) [inline packages/go/spec/docs.go:1]: I feel like this is just a bad GitHub artifact, but it's weird that it thinks a typescript file got moved to a go file. Does this make any sense?

File: packages/go/spec/go.mod
Comment (juggernot325) [inline packages/go/spec/go.mod:1]: This looks to be another confused file move.

**Review by mistahj67 — COMMENTED** [review]
~All of the `.yarn` files should be git ignored right?~
Not ignored to pin versions

File: packages/go/spec/v2/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,833 @@
...
+
+    Finally, base64 encode the final hash and write the three required headers before
+    sending the request:
+    ```python
+    encoded_hash = base64_encode(digester.hash())
+    request.header.write('Authorization', 'bhesignature ' + token_id)
+    request.header.write('RequestDate', request_datetime)
+    request.header.write('Signature', encoded_hash)
+    ```
+  #termsOfService: TBD
```
Comment (mistahj67) [inline packages/go/spec/v2/openapi.yaml:129]: Did we want to leave this TBD terms of service or remove it?

File: cmd/api/src/model/auth.go [hunk 2]
```diff
@@ -488,6 +487,10 @@ func (s *User) RemoveRole(role Role) {
 	s.Roles = s.Roles.RemoveByName(role.Name)
 }
 
+type Sortable interface {
```
Comment (codydbentley) [inline cmd/api/src/model/auth.go:490]: remove this

File: justfile [hunk 3]
```diff
@@ -194,3 +194,7 @@ run-bhce-container platform='linux/amd64' tag='custom' version='v5.0.0' *ARGS=''
 
 check-license:
   python3 license_check.py
+
```
Comment (codydbentley) [inline justfile:197]: clean up

**Comment by codydbentley** [comment]
too much "trust me bro" in my approach, will be breaking this up into digestible chunks

**Pull request closed without merging**